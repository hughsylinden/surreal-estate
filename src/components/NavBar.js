import React from "react";
import "../styles/NavBar.css";
import { Link } from "react-router-dom";
// import logo from "../styles/img/houselogo.svg";
// <img src={logo} alt="logo" className="navbar-logo" />

const NavBar = () => {
  return (
    <div className="navbar">
      <ul className="navbar-links">
        <li className="navbar-links-item">
          <Link to="/">properties</Link>
        </li>
        <li className="navbar-links-item">
          <Link to="/add-property">add property</Link>
        </li>
      </ul>
    </div>
  );
};

export default NavBar;
