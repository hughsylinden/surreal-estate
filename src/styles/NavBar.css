.navbar{
  border-color: #4D5057;
  background-color: #F1E8B8;
  padding-top:10px;
  padding-bottom:10px;
  border-bottom: solid 1px #393D3F;
}

.navbar-logo{
  width: 300px;
}

.navbar-links-item{
  list-style: none;
  display: inline-block;
  padding-right: 20px;
}

.navbar-links{
  margin:0;
}