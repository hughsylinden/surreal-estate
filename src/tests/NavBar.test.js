/* eslint-disable no-unused-vars */
import { render, screen } from "@testing-library/react";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  test("it renders a NavBar component", () => {
    const { asFragment } = render(
      <Router>
        <NavBar />
      </Router>
    );
    expect(asFragment).toMatchSnapshot();
  });

  test("it renders properties and add properties elements", () => {
    render(
      <Router>
        <NavBar />
      </Router>
    );
    const properties = screen.getByText("properties");
    const addProperty = screen.getByText("add property");
    expect(properties).toBeInTheDocument();
    expect(addProperty).toBeInTheDocument();
  });
});
