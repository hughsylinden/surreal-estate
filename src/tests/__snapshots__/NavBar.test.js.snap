// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar it renders a NavBar component 1`] = `[Function]`;
